src/feed/impl/TransactionFeed.cpp

```diff
@@ -275,9 +275,14 @@ TransactionFeed::pub(
                 if ((data != nullptr) && data->isFieldPresent(ripple::sfTakerPays) &&
                     data->isFieldPresent(ripple::sfTakerGets)) {
                     // determine the OrderBook
+                    std::optional<ripple::uint256> domainID;
+                    if (data->isFieldPresent(ripple::sfDomainID)) {
+                        domainID = data->getFieldH256(ripple::sfDomainID);
+                    }
                     ripple::Book const book{
                         data->getFieldAmount(ripple::sfTakerGets).issue(),
-                        data->getFieldAmount(ripple::sfTakerPays).issue()
+                        data->getFieldAmount(ripple::sfTakerPays).issue(),
+                        domainID
```

Comment:
nit: we can avoid a new var by doing `(*data)[~ripple::sfDomainID]`, where `~` retreives the field as an `optional`

Reply:
Thanks, fixed!